src/ImageSharp/Processing/Processors/CLoneingImageProcessor.cs

```diff
@@ -0,0 +1,166 @@
+﻿// <copyright file="CloneingImageProcessor.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp.Processing
+{
+    using System;
+    using System.Threading.Tasks;
+
+    using ImageSharp.PixelFormats;
+    using SixLabors.Primitives;
+
+    /// <summary>
+    /// Allows the application of processors to images.
+    /// </summary>
+    /// <typeparam name="TPixel">The pixel format.</typeparam>
+    internal abstract class CloneingImageProcessor<TPixel> : IImageProcessor<TPixel>, ICloningImageProcessor<TPixel>
```

Comment:
there is no duplication really, they both follow the same patterns but the code is different, one takes 2 images the other takes 1. Basically `CloneingImageProcessor` is for image processors that require a 2nd buffer to apply changes to before overwriting the pixel data in the original. `ImageProcessor` if for processors that don't require the 2nd buffer and can apply there changes directly to the source.